met_connectors/src/frost/fetch.rs

```diff
@@ -23,33 +24,39 @@ fn extract_data(
     let data = ts_portion
         .iter_mut()
         .map(|ts| {
-            // TODO: this should be a Vec<FrostObs>?
-            let obs: FrostObs = serde_json::from_value(
+            let header = ts.get_mut("header").ok_or(Error::FindObs(
+                "couldn't find header field on tseries".to_string(),
+            ))?;
+
+            let station_id = util::extract_station_id(header)?;
+
+            // TODO: Should there be a location for each observation?
+            let location = util::extract_location(header, time)?;
+
+            // TODO: differentiate actual parse errors from missing duration?
+            let ts_time_resolution_result = util::extract_duration(header);
+            if ts_time_resolution_result.is_err()
+                || ts_time_resolution_result.unwrap() != request_time_resolution
+            {
+                return Ok(None);
+            }
+
+            let obs: Vec<FrostObs> = serde_json::from_value(
                 ts.get_mut("observations")
                     .ok_or(Error::FindObs(
                         "couldn't find observations field on tseries".to_string(),
                     ))?
-                    .get_mut(0)
-                    .ok_or(Error::FindObs(
-                        "couldn't find first observation".to_string(),
-                    ))?
                     .take(),
             )?;
 
-            let header = ts.get_mut("header").ok_or(Error::FindObs(
-                "couldn't find header field on tseries".to_string(),
-            ))?;
-            let location = util::extract_location(header, time)?;
-
-            // default to one hour if `timeseries` section is missing in the metadata
-            // TODO: we might not need this inside here, since we want all stations
-            // to have the same time resolution
-            let time_resolution =
-                util::extract_duration(header).unwrap_or(RelativeDuration::hours(1));
-
-            Ok((obs, location, time_resolution))
+            Ok(Some(((station_id, obs), location)))
         })
-        .collect::<Result<Vec<(FrostObs, FrostLatLonElev, RelativeDuration)>, Error>>()?;
+        // Is there some smart way to avoid a double collect without making the error handling
+        // messy?
+        .collect::<Result<Vec<Option<((String, Vec<FrostObs>), FrostLatLonElev)>>, Error>>()?
+        .into_iter()
+        .flatten()
+        .collect();
```

Comment:
```suggestion
        .filter_map(Result::transpose)
        .collect::<Result<Vec<((String, Vec<FrostObs>), FrostLatLonElev)>, Error>>()?;
```
But maybe a for loop works better here?

Reply:
I like your version, done.